src/openforms/appointments/views.py

```diff
@@ -41,3 +47,30 @@ def get_redirect_url(self, submission_uuid: uuid, token: str, *args, **kwargs):
             }
         )
         return f.url
+
+
+class VerifyChangeAppointmentLinkView(RedirectView):
+    def get_redirect_url(self, submission_uuid: uuid, token: str, *args, **kwargs):
+        submission = validate_url_and_get_submission(submission_uuid, token)
+
+        new_submission = submission.copy()
```

Comment:
this sould probably be a deep copy, since the dicts in JSONFields are still the same instances and if we were to mutate these and (accidentally) save the original submission, this would now incorrectly be updated too.